test/end2end_test.go

```diff
@@ -5351,6 +5351,31 @@ func testRPCTimeout(t *testing.T, e env) {
 	}
 }
 
+// Tests that the client doesn't send a negative timeout to the server. If the
+// server receives a negative timeout, it would return an internal status. The
+// client checks the context error before starting a stream, however the context
+// may expire after this check and before the timeout is calculated.
+func (s) TestNegativeRPCTimeout(t *testing.T) {
+	server := stubserver.StartTestService(t, nil)
+	defer server.Stop()
+
+	if err := server.StartClient(); err != nil {
+		t.Fatal(err)
```

Comment:
This should say what it's doing in the error message, to make debugging a failure easier.